packages/flutter_tools/lib/src/test/coverage_collector.dart

```diff
@@ -12,57 +12,83 @@ import '../base/io.dart';
 import '../dart/package_map.dart';
 import '../globals.dart';
 
+/// A class that's used to collect coverage data during tests.
 class CoverageCollector {
-  static final CoverageCollector instance = new CoverageCollector();
+  /// The singleton instance of the coverage collector.
+  static final CoverageCollector instance = new CoverageCollector._();
 
+  CoverageCollector._();
+
+  /// By default, coverage collection is not enabled. Set [enabled] to true
+  /// to turn on coverage collection.
   bool enabled = false;
   int observatoryPort;
 
-  void collectCoverage({
+  /// Adds a coverage collection tasks to the pending queue. The task will not
+  /// begin collecting coverage data until [CoverageCollectionTask.start] is
+  /// called.
+  ///
+  /// This should be called after a process has been started so that this
```

Comment:
maybe add that it's not so much that it should be added after that process is created so much as before the process has finished doing its work? Not sure how to phrase it.